app/models/services/user_provided_service_instance.rb

```diff
@@ -8,6 +8,18 @@ class UserProvidedServiceInstance < ServiceInstance
 
     add_association_dependencies service_bindings: :destroy
 
+    def after_update
```

Comment:
I know it seems like there is an existing pattern of doing this in the after update, and to be fair I'm not too familiar with the services corners of the API and what patterns exist there that we wish to propagate forward, but I do know that a stylistic change we tried to push forward in the V3 API in general was to move away from relying on async model hooks (see these bullet point 3 from these [notes](https://github.com/cloudfoundry/cloud_controller_ng/wiki/Notes-on-V3-Architecture#preamble)).

What are your thoughts on moving this action to the update action, maybe in this [transaction](https://github.com/cloudfoundry/cloud_controller_ng/blob/4c355140498a284662f589bbac4efc920ab2788f/app/actions/service_instance_update_user_provided.rb#L30-L34), using the updates defined [here](https://github.com/cloudfoundry/cloud_controller_ng/blob/4c355140498a284662f589bbac4efc920ab2788f/app/actions/service_instance_update_user_provided.rb#L19-L20) as a conditional.

Reply:
Ah I hadn't seen that doc. I was going for consistency with v2 but I guess since it uses a bit of an anti-pattern we are moving away from that! I've pushed a change doing as recommended and removed the other commit. This leaves the route_service_url update not working in v2 but it has been like that for a long time and getting this v3 endpoint working (esp. for credentials) is more important right now I think